src/index.js

```diff
@@ -200,6 +192,62 @@ class FloodSub extends EventEmitter {
   }
 
   /**
+   * Mounts the floodsub protocol onto the libp2p node and sends our
+   * subscriptions to every peer conneceted
+   *
+   * @param {Function} callback
+   * @returns {undefined}
+   *
+   */
+  start (callback) {
+    if (this.started) {
+      return setImmediate(() => callback(new Error('already started')))
+    }
+
+    this.libp2p.handle(multicodec, this._onConnection)
+
+    // Speed up any new peer that comes in my way
+    this.libp2p.swarm.on('peer-mux-established', this._dialPeer)
+
+    // Dial already connected peers
+    const peerInfos = values(this.libp2p.peerBook.getAll())
+
+    peerInfos.forEach((peerInfo) => {
+      this._dialPeer(peerInfo)
+    })
+
+    setImmediate(() => {
+      this.started = true
+      callback()
```

Comment:
mostly future proofing, but I was also asking myself if we should wait for the dials to happen.